src/resources/views/crud/fields/relationship/fetch_or_create.blade.php

```diff
@@ -0,0 +1,558 @@
+<!--  relationship  -->
+
+@php
+    use Illuminate\Support\Str;
+
+    //in case entity is superNews we want the url friendly super-news
+    $routeEntity = Str::kebab($field['entity']);
+
+    $connected_entity = new $field['model'];
+    $connected_entity_key_name = $connected_entity->getKeyName();
+
+    //we need to explicitly define field type to load correct assets
```

Comment:
Hmmm.... I think overwriting the field type is better done wherever the switchboard logic is. If that's where the source of the problem is, that's also where the solution should be.